roxie/ccd/ccdqueue.cpp

```diff
@@ -746,15 +745,14 @@ class RoxieQueue : public CInterface, implements IThreadFactory
     CriticalSection qcrit;
     unsigned headRegionSize;
     unsigned numWorkers;
-    unsigned started;
+    RelaxedAtomic<unsigned> started;
     std::atomic<unsigned> idle;
 
     void noteQueued()
     {
-        CriticalBlock b(counterCrit);
         queueLength++;
         if (queueLength>maxQueueLength)
-            maxQueueLength = queueLength;
+            maxQueueLength.store(queueLength);
```

Comment:
same issue as elsewhere ensuring that maximum is consistently updated.  (And value should probably be in a local variable s the result of ++queryLength.)